core/Model/CalendarModel.vala

```diff
@@ -294,6 +290,36 @@ public class Maya.Model.CalendarModel : Object {
 
     //--- Helper Methods ---//
 
+    /** Set the week_starts_on property: the first day of the week.
+     *
+     * Locale handling is based on information from
+     * https://sourceware.org/glibc/wiki/Locales
+     */
+    private GLib.DateWeekday get_week_start () {
+        // Set the "baseline" for start of week: Sunday or Monday?
+        uint week_day1 = (uint) Posix.NLTime.WEEK_1STDAY.to_string ();
```

Comment:
Casting a **string** to a **uint** looks strange.  Shouldn't you use `int.parse ()`?  Or just compare strings?

Reply:
See my comment on the other parse suggestion. This is again the only way I could get this to compile/run. I think that what WEEK_FIRSTDAY produces is just a word in the union, and the string pointer is null. (maybe?) 

When I try to actually use the string, it uses the resulting `19971130` as the pointer address to the string array, which segfaults of course. So the only way I can figure out to not have memory issues is to take the data in memory and force Vala to use it as a number instead of an address, without ever actually referencing that address. That's why I used the cast. I just copied it from the defunct [California app](https://github.com/blankon-packages/california/blob/b3dc12893c012035fcd4fc83b383cdf8933b80f4/src/calendar/calendar-system.vala#L257), I have no idea what it's really doing (and neither do they).

Reply:
See my response above.  The only problem may be that this feels like the kind of thing that may change with a newer version of Vala, but I guess we can cross that bridge when we come to it.

Reply:
Good point. I hope it doesn't break, but if we have to stick with this method we can deal with that when we come to it. I can add a test to make sure it doesn't break in the future. I'll also add a comment to call out the weird behavior, since as you say the syntax doesn't make clear what's going on.

Reply:
Adding a test would be excellent.